WcaOnRails/app/models/competition.rb

```diff
@@ -2011,4 +2011,28 @@ def dues_per_competitor_in_usd
     dues = DuesCalculator.dues_per_competitor_in_usd(self.country_iso2, self.base_entry_fee_lowest_denomination.to_i, self.currency_code)
     dues.present? ? dues : 0
   end
+
+  def dues_payer
+    country_id = self.country.id
+    country_redirect = WfcDuesRedirect.find_by(redirect_from_country_id: country_id)
+    if country_redirect.present?
+      return WfcXeroUser.find_by(id: country_redirect.redirect_to_id)
+    end
```

Comment:
Based on the assumption of polymorphic reference (see code comment in the migration) you need to change the `country.rb` model as follows:

```ruby
  has_one :wfc_dues_redirect, as: :redirect_source, optional: true # where :redirect_source has to match the `belongs_to` association in the WfcDuesRedirect model
```

and then you can simply do as follows...

```suggestion
  if self.country.wfc_dues_redirect.present?
    return self.country.wfc_dues_redirect.redirect_to
  end
```

Reply:
Done